angular/projects/spark-angular/src/lib/components/sprk-autocomplete/sprk-autocomplete-result/sprk-autocomplete-result.directive.ts

```diff
@@ -0,0 +1,154 @@
+import {
+  Directive,
+  ElementRef,
+  OnInit,
+  HostListener,
+  Output,
+  EventEmitter,
+  Input,
+  HostBinding,
+  Renderer2,
+} from '@angular/core';
+import { uniqueId } from 'lodash';
+
+@Directive({
+  selector: '[sprkAutocompleteResult]',
+})
+export class SprkAutocompleteResultDirective implements OnInit {
+  /**
+   * @ignore
+   */
+  constructor(public ref: ElementRef, private renderer: Renderer2) {}
+
+  /**
+   * The value supplied will be assigned
+   * to the `data-id` attribute on the
+   * element. This is intended to be
+   * used as a selector for automated
+   * tools. This value should be unique
+   * per page.
+   */
+  @HostBinding('attr.data-id')
+  @Input()
+  idString: string;
+
+  /**
+   * The value supplied will be assigned to the
+   * `data-analytics` attribute on the element.
+   * Intended for an outside
+   * library to capture data.
+   */
+  @HostBinding('attr.data-analytics')
+  @Input()
+  analyticsString: string;
+
+  /**
+   * Expects a space separated string
+   * of classes to be added to the
+   * element.
+   */
+  @Input()
+  additionalClasses: string;
+
+  /**
+   * @ignore
+   *
+   */
+  private _isSelected = false;
+
+  /**
+   * If true, this Autocomplete result will be highlighted.
+   */
+  @HostBinding('class.sprk-c-Autocomplete__result--active')
+  @HostBinding('attr.aria-selected')
+  @Input()
+  set isSelected(value: boolean) {
+    /**
+     * Normally Spark would put this kind of behavior in ngOnChanges, but
+     * that lifecycle event is not triggered when an Input is updated directly
+     * from a parent component. This logic is moved into a setter on the Input
+     * so that it is triggered when sprk-autocomplete changes this value.
+     */
+    this._isSelected = value;
+
+    if (this._isSelected) {
+      var result = this.ref.nativeElement;
```

Comment:
I don't see `result` getting set to a new value here, can update this to be `const`